lib/post_destroyer.rb

```diff
@@ -254,7 +254,7 @@ def make_previous_post_the_last_one
       .limit(1)
       .first
 
-    if last_post.present? && @post.topic.present?
+    if last_post.present? && @post.topic.present? && @post.topic.persisted?
```

Comment:
Is this related to the fix?

If it is, I think you can drop the `@post.topic.present?` part - the only caller (`perform_delete`) is already checking for topic (`if @post.topic`) before calling this method.

Reply:
Nice catch! yeah, actually both of them are not needed so I dropped them :)